FlightJournal.Web/Controllers/ReportController.cs

```diff
@@ -51,6 +51,11 @@ public ActionResult Index(DateTime? date, string club = null)
 
                         rptYear.DistinctLocations = rptYear.Flights.Select(d => d.StartedFrom).Distinct().OrderBy(d => d.Name);
 
+                        foreach (var flight in rptYear.Flights)
+                        {
+                            flight.HasTrainingData = db.AppliedExercises.Any(x => x.FlightId == flight.FlightId);
```

Comment:
Denne når jeg frem til bliver en per flyvning lookup i databasen?

Man kunne overveje at gemme den konkrete status som bool i forbindelse med redigeringen af excercise for at holde LINQ laget enkelt. 
En anden model er at få en left outer join ind på AppliedExercises i det første LINQ kald på flights (måske ender vi i custom sql... )

Reply:
Ja, det er nok en del billigere at hente alle AppliedExercises og lave opslaget i memory.

Jeg har hidtil prøvet at påvirke 'gamle' dele af modellen (som ikke har noget med skoleflyvning at gøre) mest muligt, men det kan godt være at det ikke holder i længden.